### PR TITLE
Refactor codegen to output multiple versions.

### PR DESCRIPTION
**Pull request by simeoncran**
This does not output multiple versions - that is coming in a subsequent change.

### CODE REVIEW

File: source/UIData/CodeGen/AnimatedVisualSourceInfo.cs [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+            LoadedImageSurfaceNodes = loadedImageSurfaceNodes;
+        }
+
+        /// <summary>
+        /// Gets the name of the IAnimatedVisualSource class that will be exposed to users.
+        /// </summary>
+        public string ClassName { get; }
+
+        /// <summary>
+        /// Gets the name of the field in the instantiator class that hold the reusable ExpressionAnimation.
```
Comment (eliezerpMS) [inline source/UIData/CodeGen/AnimatedVisualSourceInfo.cs]: >hold [](start = 70, length = 4)

Holds

**Comment by eliezerpMS** [comment]
> #endif

Should this still be here?

---
Refers to: source/UIData/CodeGen/InstantiatorGeneratorBase.cs:301 in 6585840. [](commit_id = 6585840121acf9f6ed8e532c113272dbe020d660, deletion_comment = False)

**Comment by simeoncran** [comment]
> #endif

Yes, it's the new implementation. I hacked it out to get the tests to pass. I'll add a TODO.

---
In reply to: [545123339](https://github.com/windows-toolkit/Lottie-Windows/pull/159#issuecomment-545123339) [](ancestors = 545123339)

---
Refers to: source/UIData/CodeGen/InstantiatorGeneratorBase.cs:301 in 6585840. [](commit_id = 6585840121acf9f6ed8e532c113272dbe020d660, deletion_comment = False)

**Comment by eliezerpMS** [comment]
>             // The name that is giving to the node by the NodeNamer. This name is used to generate factory method

given

---
Refers to: source/UIData/CodeGen/InstantiatorGeneratorBase.cs:2094 in 7457303. [](commit_id = 74573033cac1f4fe500916bd70da4c8814b8fe7f, deletion_comment = False)

**Comment by eliezerpMS** [comment]
>             // True if the node describes and object that can be shared between

an

---
Refers to: source/UIData/CodeGen/InstantiatorGeneratorBase.cs:2178 in 7457303. [](commit_id = 74573033cac1f4fe500916bd70da4c8814b8fe7f, deletion_comment = False)

**Comment by eliezerpMS** [comment]
>             // Generate names for each of the canonical nodes (i.e. the first node in each group)

nit: end with "."

---
Refers to: source/UIData/CodeGen/InstantiatorGeneratorBase.cs:94 in 7457303. [](commit_id = 74573033cac1f4fe500916bd70da4c8814b8fe7f, deletion_comment = False)

**Pull request merged**